### ISSUE TITLE
Design fine-grained access control within record manager

### ISSUE THREAD

**Issue by blcham** [kbss-cvut/record-manager-ui#202]

**Comment by palagdan** [comment]
@blcham 

If I understand the task correctly, we can add two different groups for operators and suppliers, as shown in the picture below:

![Image](https://github.com/user-attachments/assets/a85adc5e-b411-4ecc-b5a0-9ee2b77c9266)

Each group can have its own subgroups that inherit roles from the parent group. This allows us to create different subgroups with specific roles.

**Comment by blcham** [comment]
Open questions:
- [x] Should we show a group of roles in the RM-UI? **Yes**
   - It makes sense, at least in the internal authorization case. Moreover, we will have at least some role names prepared with some initial semantics on how It could be used and redefined in the Keycloak. Note that the Inheritance of roles is working in groups within Keycloak.
-  [ ]  Is it possible to retrieve role groups in the token?
-  [ ] How will we treat users not being in any institution?

**Comment by palagdan** [comment]
Yes, we can retrieve groups in the token, see https://stackoverflow.com/questions/56362197/keycloak-oidc-retrieve-user-groups-attributes

**Comment by blcham** [comment]
ok, try to implement part of the code related to differences between supplier and operator

**Comment by palagdan** [comment]
@blcham 
I set up Keycloak with two groups (Operator and Supplier) and configured client scopes to include this information in the token. From the access token, I can retrieve this information. It seems to be working

![Image](https://github.com/user-attachments/assets/26044962-022e-4010-9485-cfdf97b33309)

**Comment by palagdan** [comment]
@blcham 
Regarding your second task, I have set up the groups with default roles. As shown in the picture, the supplier group has assigned roles to the user. I've only set Admin role myself.

**Comment by blcham** [comment]
- roles delete/edit/view should be 3 separate roles
- there should be global prefix to this role i.e. record_manager_ or just rm_

**Comment by palagdan** [comment]
@blcham 
1)  I want to discuss what roles should I assign to each group. 
2)  Should I change 'Roles' to 'Group' and display the Group name instead? Since a user can have multiple roles, it might be more effective to show the group to which the user is assigned.
![Image](https://github.com/user-attachments/assets/d844f1d6-1583-473f-a866-94637e976cd0)

**Comment by palagdan** [comment]
@blcham 
We currently have the following roles:

- **ROLE_ADMIN**
- **ROLE_USER**
- **rm_complete_records**
- **rm_delete_all_records**
- **rm_delete_organizations_records**
- **rm_edit_all_records**
- **rm_edit_organizations_records**
- **rm_edit_users**
- **rm_import_codelist**
- **rm_publish_records**
- **rm_reject_records**
- **rm_view_all_records**
- **rm_view_organizations_records**

If Keycloak saves personal information about users to the repository, why can't it save roles and groups as well? It doesn't make sense to handle this on the backend because, with every request, the system would need to check Keycloak for changes and then update the repository accordingly.

**Comment by palagdan** [comment]
@blcham 

The Publish button is in the Operator extension (not Supplier). Can you explain what this button does?

![Image](https://github.com/user-attachments/assets/e59c3abc-d744-42f8-bbcc-e3ac4926ed5a)

**Comment by blcham** [comment]
publish button imports all completed records from operator deployment to supplier deployment

**Comment by blcham** [comment]
First step:
- [ ] change backend to contain specific roles together with temporary role ADMINISTRATOR and USER
- [ ] change frontend to support backend
- [ ] support keycloak-auth profile
   - [ ] add code to show role group of current user
   - [ ] change kbss distribution keycloak configuration to contain new roles 
- [ ] support internal-auth profile
   - [ ] add role groups in .trig file in record-manager-ui/deploy/internal-auth

**Comment by blcham** [comment]
\kbss-bot help

**Comment by blcham** [comment]
\kbss

**Comment by blcham** [comment]
\kbss help

**Comment by blcham** [comment]
@blcham should test removal of supplier/operator from `EXTENSIONS: "${RECORD_MANAGER_EXTENSIONS....` to close this issue.
